src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -26,6 +26,12 @@ def get_location_completion_list(prefix, **kwargs):#pylint: disable=unused-argum
     result = get_subscription_locations()
     return [l.name for l in result]
 
+def location_name_type(name):
+    if ' ' in name:
+        # if display name is provided, attempt to convert to short form name
+        name = next((l.name for l in get_subscription_locations() if l.display_name == name), name)
```

Comment:
Good point @derekbekoe. Since I've merged this PR, I'll do this in the case insensitive choice lists PR.